src/libraries/System.IO.FileSystem/tests/FileStream/FileStreamConformanceTests.Windows.cs

```diff
@@ -176,4 +176,148 @@ public struct SHARE_INFO_502
+                deviceInfoData.cbSize = (uint)Marshal.SizeOf(deviceInfoData);
+
+                uint deviceIndex = 0;
+                while (SetupDiEnumDeviceInfo(deviceInfoSet, deviceIndex++, ref deviceInfoData))
+                {
+                    if (Marshal.GetLastWin32Error() == ERROR_NO_MORE_ITEMS)
+                    {
+                        break;
+                    }
+
+                    SP_DEVICE_INTERFACE_DATA deviceInterfaceData = new SP_DEVICE_INTERFACE_DATA();
+                    deviceInterfaceData.cbSize = Marshal.SizeOf(deviceInterfaceData);
+
+                    if (!SetupDiEnumDeviceInterfaces(deviceInfoSet, IntPtr.Zero, in HidGuid, deviceIndex, ref deviceInterfaceData))
+                    {
+                        continue;
+                    }
+
+                    SP_DEVICE_INTERFACE_DETAIL_DATA deviceInterfaceDetailData = new SP_DEVICE_INTERFACE_DETAIL_DATA();
+                    deviceInterfaceDetailData.cbSize = IntPtr.Size == 8 ? 8 : 6;
+
+                    uint size = (uint)Marshal.SizeOf(deviceInterfaceDetailData);
+
+                    if (!SetupDiGetDeviceInterfaceDetail(deviceInfoSet, ref deviceInterfaceData, ref deviceInterfaceDetailData, size, ref size, IntPtr.Zero))
+                    {
+                        continue;
+                    }
+
+                    string devicePath = deviceInterfaceDetailData.DevicePath;
+                    Assert.StartsWith(@"\\?\hid", devicePath);
+
+                    try
+                    {
+                        return new FileStream(devicePath, FileMode.Open, FileAccess.Read, FileShare.Read, 0, FileOptions.Asynchronous);
+                    }
+                    catch (IOException)
+                    {
+                        continue; // device has been locked by another process
+                    }
+                }
+            }
+            finally
+            {
+                SetupDiDestroyDeviceInfoList(deviceInfoSet);
+            }
+
+            return null;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        struct SP_DEVICE_INTERFACE_DATA
+        {
+            public int cbSize;
+            public Guid interfaceClassGuid;
+            public int flags;
+            private nuint reserved;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        struct SP_DEVINFO_DATA
+        {
+            public uint cbSize;
+            public Guid ClassGuid;
+            public uint DevInst;
+            public nint Reserved;
+        }
+
+        [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Unicode)]
+        struct SP_DEVICE_INTERFACE_DETAIL_DATA
+        {
+            public int cbSize;
+            [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 256)] // 256 should be always enough for device interface path
+            public string DevicePath;
+        }
+
+        [DllImport("hid.dll", SetLastError = true)]
+        static extern void HidD_GetHidGuid(out Guid Guid);
```

Comment:
Do we have official guidance indicating if we should keep the exact parameter names as defined in the Windows AP?

For example, the `Guid` parameter's original name is `HidGuid`:

https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/hidsdi/nf-hidsdi-hidd_gethidguid

Reply:
I don't know if we have official guidelines for that, however I have aligned the names